scapy/contrib/coap_socket.py

```diff
@@ -0,0 +1,1001 @@
+# SPDX-License-Identifier: GPL-2.0-only
+# This file is part of Scapy
+# See https://scapy.net/ for more information
+# Copyright (C) 2024 eHonnef <contact@honnef.net>
+
+# scapy.contrib.description = CoAP Socket Library / RFC 7252
+# scapy.contrib.status = library
+
+import logging
+import random
+import socket
+import time
+
+# Typing imports
+from typing import (
+    Optional,
+    Union,
+    Tuple,
+    Any,
+    cast,
+    Type
+)
+
+from scapy.error import Scapy_Exception
+from scapy.packet import Packet
+from scapy.contrib.coap import CoAP, coap_options, coap_codes
+from scapy.contrib.isotp.isotp_soft_socket import TimeoutScheduler
+from scapy.data import MTU
+from scapy.utils import EDecimal
+from scapy.automaton import ObjectPipe, select_objects
+
+from scapy.supersocket import SuperSocket, SimpleSocket
+
+"""
+CoAP message request codes (RFC 7252 @ section-5.8.1)
```

Comment:
Is it necessary to redefine the codes here, can’t you use the ones in coap.py

Reply:
Not really, I don't want to use magic numbers, so I did an "enumerator" for those, but I moved these to coap.py